pkg/api/pod/util.go

```diff
@@ -444,6 +444,14 @@ func dropDisabledFields(
 		podSpec.SetHostnameAsFQDN = nil
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) && !inPlacePodVerticalScalingInUse(oldPodSpec) {
+		// Drop ResourcesAllocated and ResizePolicy fields. Don't drop updates to Resources field because
+		// template spec Resources field is mutable for certain controllers. Let ValidatePodUpdate handle it.
+		for i := range podSpec.Containers {
```

Comment:
also initContainers/ephemeralContainers? should this use the VisitContainers helper?

Reply:
Let me try out init and ephemeral containers and be sure about them. I think initContainers would need this.